python/lsst/ts/salobj/configurable_csc.py

```diff
@@ -131,18 +131,18 @@ def __init__(
         initial_simulation_mode=0,
     ):
 
-        if not pathlib.Path(schema_path).is_file():
-            raise ValueError(f"schema_path={schema_path} is not a file")
-        with open(schema_path, "r") as f:
-            schema_data = f.read()
         try:
-            schema = yaml.safe_load(schema_data)
+            with open(schema_path, "r") as f:
+                schema = yaml.safe_load(f.read())
             self.config_validator = DefaultingValidator(schema=schema)
             title = schema["title"]
             name_version = title.rsplit(" ", 1)
             if len(name_version) != 2 or not name_version[1].startswith("v"):
                 raise ValueError(f"Schema title {title!r} must end with ' v<version>'")
             self.schema_version = name_version[1]
+
+        except OSError as e:
+            raise ValueError(f"While reading schema_path={schema_path}: {e}")
```

Comment:
I strongly suggest a message such as `f"Could not read schema {schema_path}: {e}"`
as I think it would be clearer.

Reply:
Also, if you are willing, please fix an existing error: schema_path is described as a "str" in the doc string, but it should be described as:
```
schema_path : `str` or `pathlib.Path`
```
It is possible to be even more general, but the above is what I have been using.